ui/src/app/views/concept-homepage/component.ts

```diff
@@ -105,36 +129,51 @@ export class ConceptHomepageComponent implements OnInit {
     } else {
       standardConceptFilter = StandardConceptFilter.ALLCONCEPTS;
     }
+    this.completedDomainSearches = [];
 
     let numCalls = 0;
-    const numCallsSubject = new BehaviorSubject<number>(0);
+    const numCallsSubject = new Subject<Domain>();
     const numCallsSubject$ = numCallsSubject.asObservable();
 
-    numCallsSubject$.subscribe((finishedCalls) => {
-      if (finishedCalls === this.conceptsCache.length) {
-        this.searchLoading = false;
-        this.setConcepts();
-      }
+    numCallsSubject$.subscribe((finishedCall) => {
+      this.completedDomainSearches.push(finishedCall);
     });
 
     this.conceptsCache.forEach((conceptDomain) => {
+      const activeTabSearch = conceptDomain.domain === this.selectedDomain;
       const request = {
         query: this.currentSearchString,
         standardConceptFilter: standardConceptFilter,
-        domain: conceptDomain.domain
+        domain: conceptDomain.domain,
+        includeDomainCounts: activeTabSearch,
+        includeVocabularyCounts: true
       };
       this.conceptsService.searchConcepts(this.wsNamespace, this.wsId, request)
         .subscribe((response) => {
           numCalls += 1;
-          numCallsSubject.next(numCalls);
+          numCallsSubject.next(conceptDomain.domain);
```

Comment:
Not really part of the PR but I'm not sure I'm following this `numCallsSubject` approach, I assume it has to do with tracking multiple parallel concept requests. Did you consider using `forkJoin`?

Reply:
It is handling multiple concept requests. I think `forkJoin` would have worked before, but now it is changed to enable tabs as the data comes back, so I am not blocking on all of them. I can add a comment if you would like?